src/core/viz/expressions/torque.js

```diff
@@ -203,10 +210,99 @@ export class Torque extends BaseExpression {
         return new Date(tmix);
     }
 
+    /**
+     * Set the time stamp of the simulation
+     * @api
+     * @memberof carto.expressions.Torque
+     * @instance
+     * @name getSimTime
+     * @param {Date} simulationTime - A javascript Date object with the new simulation time
+     */
+    setSimTime(simulationTime) {
+        const tmin = this._input.min.eval();
+        const tmax = this._input.max.eval();
+        if (!isDate(simulationTime)) {
+            throw new TypeError(`torque.setSimTime requires a valid date as parameter but got ${simulationTime}`);
+        }
+        if (simulationTime.getTime() < tmin) {
+            throw new RangeError('torque.setSimTime requires the date parameter to be higher than the minimal date of the dataset');
+        }
+        if (simulationTime.getTime() > tmax) {
+            throw new RangeError('torque.setSimTime requires the date parameter to be lower than the maximun date of the dataset');
+        }
+        this.progress.expr = (simulationTime.getTime() - tmin) / (tmax - tmin);
+    }
+
+    /**
+     * Get the simulation progress.
+     * 
+     * @returns {Number} A number representing the progress. 0 when the animation just started and 1 when complete.
+     * @api
+     * @instance
+     * @memberof carto.expressions.Torque
+     * @name getSimTime
+     */
+    getSimProgress() {
+        return this.progress.value;
+    }
+
+    /**
+     * Set the simulation progress from 0 to 1.
+     * @param {number} progress - A number in the [0-1] range setting the animation progress.
+     * @api
+     * @instance
+     * @memberof carto.expressions.Torque
+     * @name getSimTime
+     */
+    setSimProgress(progress) {
+        progress = Number.parseFloat(progress);
```

Comment:
parseFloat? Why?

Reply:
to ensure the parameter is a number and make the `progress < 0 || progress > 1` comparison safe.

we could also check the type and throw an error. what do you prefer?

Reply:
Check the type, please, there is an util for that.